jujugui/static/gui/src/app/app.js

```diff
@@ -561,7 +561,11 @@ YUI.add('juju-gui', function(Y) {
       // arrive with successful authentication), and redispatch.
       this.env.after('connectedChange', function(ev) {
         if (ev.newVal === true) {
-          this.db.reset();
+          // If we're in gisf we do not want to empty the db when we connect
+          // because the user may have made changes to the temporary model.
+          if (!this.get('gisf')) {
```

Comment:
Typically, when switching models we want to reset the db so that we are starting from scratch when all of the new deltas come in. In the case where we haven't yet connected to any model yet we do not want to empty out the db because the ecs and db rely on each other for any changes that still need to be applied to the model. 

This means that we will need a flag to indicate that, not only are we in gisf, but that we're switching models and not just haven't yet connected or we'll introduce a bug switching models here.